modules/programs/swayr.nix

```diff
@@ -0,0 +1,135 @@
+          container_format = "{indent}<b>Container [{layout}]</b> <i>{marks}</i> on workspace {workspace_name}    <span alpha=\"20000\">({id})</span>";
+          window_format = "img:{app_icon}:text:{indent}<i>{app_name}</i> — {urgency_start}<b>“{title}”</b>{urgency_end} <i>{marks}</i> on workspace {workspace_name} / {output_name}    <span alpha=\"20000\">({id})</span>";
+          indent = "    ";
+          urgency_start = "<span background=\"darkred\" foreground=\"yellow\">";
+          urgency_end = "</span>";
+          html_escape = true;
+        };
+
+        layout = {
+          auto_tile = false;
+          auto_tile_min_window_width_per_output_width = [
+            [ 800 400 ]
+            [ 1024 500 ]
+            [ 1280 600 ]
+            [ 1400 680 ]
+            [ 1440 700 ]
+            [ 1600 780 ]
+            [ 1680 780 ]
+            [ 1920 920 ]
+            [ 2048 980 ]
+            [ 2560 1000 ]
+            [ 3440 1200 ]
+            [ 3840 1280 ]
+            [ 4096 1400 ]
+            [ 4480 1600 ]
+            [ 7680 2400 ]
+          ];
+        };
+
+        focus = {
+          lockin_delay = 750;
+        };
+
+        misc = {
+          seq_inhibit = false;
+        };
+      '';
+      description = ''
+        Configuration included in `config.toml`.
+        For available options see <https://git.sr.ht/~tsdh/swayr#swayr-configuration>
+      '';
+    };
+
+    extraConfig = mkOption {
+      type = types.lines;
+      default = "";
+      description = ''
+        Extra configuration lines to append to the swayr
+        configuration file.
+      '';
+    };
+
+    systemd.enable = mkEnableOption "swayr systemd integration";
+    systemd.target = mkOption {
+      type = types.str;
+      default = "graphical-session.target";
+      description = ''
+        Systemd target to bind to.
+      '';
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.swayr;
+      defaultText = literalExpression "pkgs.swayr";
+      description = "swayr package to use.";
+    };
+  };
+
+  config = mkIf cfg.enable (mkMerge [
+    {
+      home.packages = [ cfg.package ];
+
+      # Creating an empty file on empty configuration is desirable, otherwise swayrd will create the file on startup.
+      xdg.configFile."swayr/config.toml" =
+        mkIf (cfg.settings != { }) { source = finalConfig; };
+    }
+
+    (mkIf cfg.systemd.enable {
+      systemd.user.services.swayrd = {
+        Unit = {
+          Description = "A window-switcher & more for sway";
+          Documentation = "https://sr.ht/~tsdh/swayr";
+          After = [ cfg.systemd.target ];
+          PartOf = [ cfg.systemd.target ];
+          X-Restart-Triggers = mkIf (cfg.settings != { })
+            [ "${config.xdg.configFile."swayr/config.toml".source}" ];
+        };
+        Service = {
+          Environment = "RUST_BACKTRACE=1";
```

Comment:
I tried to add fuzzel to the service PATH with 
```
  systemd.user.services.swayrd.Service = {
   Environment = [ "PATH=${lib.makeBinPath [ pkgs.fuzzel pkgs.wofi ]}" 
   ];
  };
```
which fails with
```
error: The option `home-manager.users.teto.systemd.user.services.swayrd.Service.Environment' has conflicting definition values:
       - In `/nix/store/2p117prwd40gzsnxg8823xskmgkpsaq5-source/hm/profiles/sway.nix': "PATH=/nix/store/1kg8yaz49vxy753dkshzlbcclx2v5g7l-fuzzel-1.9.2/bin:/nix/store/qhg92a8hd0gr8c5yjxw0nkn6xw402qj8-wofi-1.3/bin"
       - In `/nix/store/55y5rag38lyw9sb4d5wwdlcjh52ivkl0-source/modules/programs/swayr.nix': "RUST_BACKTRACE=1"
```
Can you set `Environment = [ "RUST_BACKTRACE=1" ];` instead so the module can merge the different values please ? tested with success locally.